server/src/main.rs

```diff
@@ -0,0 +1,41 @@
+// WIP: not ready for reviews
+
+pub mod service;
+
+use std::collections::HashMap;
+
+use once_cell::sync::OnceCell;
+use oo7::portal::Keyring;
+use std::future::pending;
+use zbus::{ConnectionBuilder, Result};
+
+use crate::service::collection::init_collection_service;
+use crate::service::service::init_secret_service;
+use crate::service::session::{init_session_service, Session};
```

Comment:
This should not be done here at all; you just need to initialize the service & wait until an interaction happens before initializing anything else

Reply:
I'm a bit unclear about this :) Is this only for `init_session_service`

Reply:
All of them, there shouldn't be any initialization done from the functions.

Reply:
I mean there would be, but indirectly through constructing a Service would construct the types it needs to function.

Reply:
Done, thanks!